File: app/src/main/java/me/sunzheng/mana/LaunchActivity.java
package me.sunzheng.mana;

import android.app.Activity;
import android.content.Context;
import android.content.Intent;
import android.content.SharedPreferences;
import android.os.Bundle;

import me.sunzheng.mana.utils.PreferenceManager;

public class LaunchActivity extends Activity {
    SharedPreferences sharedPreferences;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        sharedPreferences = getSharedPreferences(PreferenceManager.Global.STR_SP_NAME, Context.MODE_PRIVATE);
        Intent intent = null;
        if (sharedPreferences.getBoolean(PreferenceManager.Global.BOOL_IS_REMEMBERD, false)) {
            intent = new Intent(this, MainActivity.class);
        } else {
            intent = new Intent(this, LoginActivity.class);
        }
        startActivity(intent);
        finish();
    }
}

File: app/src/main/java/me/sunzheng/mana/LaunchActivity.kt
package me.sunzheng.mana

import android.app.Activity
import android.content.Intent
import android.content.SharedPreferences
import android.os.Bundle
import me.sunzheng.mana.MainActivity
import me.sunzheng.mana.utils.PreferenceManager

class LaunchActivity : Activity() {
    lateinit var sharedPreferences: SharedPreferences
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        sharedPreferences = getSharedPreferences(PreferenceManager.Global.STR_SP_NAME, MODE_PRIVATE)
        var intent: Intent? = if (sharedPreferences.getBoolean(
                PreferenceManager.Global.BOOL_IS_REMEMBERD,
                false
            )
        ) {
            Intent(this, MainActivity::class.java)
        } else {
            Intent(this, LoginActivity::class.java)
        }
        startActivity(intent)
        finish()
    }
}